.github/actions/setup-env/action.yaml

```diff
@@ -0,0 +1,22 @@
+name: Setup Environment
+description: 'Setup Environment for workflows'
+
+inputs:
+  go:
+    description: 'Install Golang'
+    required: false
+    default: 'false'
+  go-cache:
+    description: 'Cache Golang dependencies'
+    required: false
+    default: 'true'
+
```

Comment:
Add a required flag for the `go.mod` file as we can pass this to the setup step, which will then pull the correct go version. Should prevent this workflow from getting out-of-sync with the project's go version
```yaml
  go-version-file:
    description: 'Path to go.mod to pull go version from'
    required: true
# ...
    - name: Set up Go
      uses: actions/setup-go@v5
      with:
        cache: ${{ inputs.go-cache }}
        go-version-file: ${{ inputs.go-version-file }}
```

Reply:
done